### PR TITLE
[NDS-986] feat: new table, hooks and subcomponents

### PR DESCRIPTION
**Pull request by kostasdano**
## Description

[NDS-986](https://orfium.atlassian.net/browse/NDS-986)

This PR introduces the **new Table component**, the **useTable hook** and the **subcomponents**.

This PR is more of a base for the next features to be implemented; the only variation of the Table implemented here is:
- Row and Col sizing

[NDS-986]: https://orfium.atlassian.net/browse/NDS-986?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/Table/Table.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+              </TR>
+            );
+          })}
+        </TBody>
+      </table>
+    </div>
+  );
+};
+
+export default Table;
```
Comment (panvourtsis) [inline src/components/Table/Table.tsx:48]: small note for the future, i think this and others here must be memoized

File: src/components/Table/components/TD/TD.style.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      return theme.globals.sizing.get('11');
+  }
+};
+
+export const tdContainer =
+  ({ rowSize }: Pick<TableProps<any>, 'rowSize'>) =>
+  (theme: Theme): SerializedStyles => {
+    return css`
+      height: ${getMinHeight(rowSize)(theme)};
+      padding: 8px 16px;
```
Comment (panvourtsis) [inline src/components/Table/components/TD/TD.style.ts:24]: this is not coming from any token?

File: src/components/Table/components/TH/TH.style.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+export const thContainer =
+  ({ rowSize, width }: Pick<TableProps<any>, 'rowSize'> & { width?: number }) =>
+  (theme: Theme): SerializedStyles => {
+    return css`
+      width: ${width ? `${width}%` : undefined};
+      height: ${getMinHeight(rowSize)(theme)};
+      align-content: center;
+      box-sizing: border-box;
+      padding: 8px 16px;
```
Comment (panvourtsis) [inline src/components/Table/components/TH/TH.style.ts:18]: same here? no token for that?

File: src/components/Table/components/TH/TH.style.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+/** @TODO replace all css with tokens */
+
+export const thContainer =
+  ({ rowSize, width }: Pick<TableProps<any>, 'rowSize'> & { width?: number }) =>
+  (theme: Theme): SerializedStyles => {
+    return css`
+      width: ${width ? `${width}%` : undefined};
+      height: ${getMinHeight(rowSize)(theme)};
+      align-content: center;
```
Comment (panvourtsis) [inline src/components/Table/components/TH/TH.style.ts:16]: doesn't seem centered on designs

File: src/components/Table/Table.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+};
+
+export const ColumnAndRowSizing = {
+  render: (args) => {
+    const { rowSize, firstNameWidth, lastNameWidth, ageWidth, jobWidth } = args;
+
+    const columns = [
+      { id: 'firstName', header: 'First Name', width: firstNameWidth },
+      { id: 'lastName', header: 'Last Name', width: lastNameWidth },
+      { id: 'age', header: 'Age', width: ageWidth },
```
Comment (panvourtsis) [inline src/components/Table/Table.stories.tsx:28]: age as number must be align right

File: src/components/Table/components/TD/TD.style.ts (hunk 2 above)
Comment (kostasdano) [reply src/components/Table/components/TD/TD.style.ts:24]: tokens will be added after #797 and the rest of the parts are merged

File: src/components/Table/components/TH/TH.style.ts (hunk 3 above)
Comment (kostasdano) [reply src/components/Table/components/TH/TH.style.ts:18]: tokens will be added after #797 and the rest of the parts are merged

**Review by panvourtsis — APPROVED** [review]

**Pull request merged**